deploy-board/deploy_board/static/js/deploy-board.js

```diff
@@ -270,3 +270,64 @@ function getDefaultPlacement(capacityCreationInfo) {
         }
     }
 }
+
+function getDefaultHostType(hostTypes, defaultHostType, defaultARMHostType) {
+    this.hostTypes = hostTypes;
+    this.defaultHostType = defaultHostType;
+    this.defaultARMHostType = defaultARMHostType;
+        var isProcessed = false;
+        var selected = false;
+        var lowestHostType = false;
+        this.checkIsDisabledHostType = function(item){
+            return item.retired || item.blessed_status === "DECOMMISSIONING";
+        };
+
+        var isSelected = function(item) {
+            return (
+                ("x86_64" === item.arch_name && item.abstract_name === this.defaultHostType)
+                || ("arm64" === item.arch_name && item.abstract_name === this.defaultARMHostType)
+                ) && !checkIsDisabledHostType(item);
```

Comment:
if there are architectures added in the future, this will fail. Is there a way to validate without hard coding or is the hardcoded part of the validation needed?